### PR TITLE
Add CSpell

### PR DESCRIPTION
**Pull request by dEdmishka**
### added cspell:
- cspell now in project!
- project-words.txt is dictionary for new words
- cspell.json is config file for cspell
- npm run cspell used for general analysis of words of src folder
- npm run cspell:check checks for typo and suggest replacement for user
- npm run cspell:apply add words to dictionary
- cspell also work in pre-commit!

### CODE REVIEW

**Review by firehawk89 — APPROVED** [review]
Looks good

**Review by DemianParkhomenko — CHANGES_REQUESTED** [review]
Please rebase and fix several comments
Good job!

good-job-2

File: .lintstagedrc.js [hunk 1]
```diff
@@ -2,4 +2,5 @@ module.exports = {
   '*': 'prettier --check --ignore-unknown',
   '*.js': 'eslint --cache',
   '*.{js,ts}': () => 'npm run typescript:check',
+  '*.{js,ts}': () => 'npm run cspell',
```
[the remark this answers is not in the document]
Comment (dEdmishka) [reply .lintstagedrc.js]: Resolved!

File: .lintstagedrc.js
[the remark this answers is not in the document]
Comment (dEdmishka) [reply .lintstagedrc.js:1]: Added new util file to src/utils folder with this code)

File: package.json [hunk 2]
```diff
@@ -14,7 +14,10 @@
     "prettier:check": "prettier --check .",
     "prettier:write": "prettier --write .",
     "typescript:check": "tsc -p tsconfig.json",
-    "prepare": "husky install"
+    "prepare": "husky install",
+    "cspell": "cspell lint \"src/**/*.js\"",
+    "cspell:apply": "cspell --words-only --unique \"src/**/*.js\" > project-words.txt",
+    "cspell:check": "cspell --no-progress --show-suggestions --show-context  \"src/**/*.js\""
```
[the remark this answers is not in the document]
Comment (dEdmishka) [reply package.json]: Shorted!

**Review by dEdmishka — COMMENTED** [review]
review is finished

**Review by DemianParkhomenko — APPROVED** [review]

**Pull request merged**